processing/src/main/java/io/druid/segment/QueryableIndexIndexableAdapter.java

```diff
@@ -229,7 +229,7 @@ public boolean hasNext()
           {
             final boolean hasNext = currRow < numRows;
             if (!hasNext && !done) {
-              CloseQuietly.close(timestamps);
+              timestamps.close();
```

Comment:
In case this throws exception, we will miss closing metrics and dims, probably we can use a Closer here ?